### DISCUSSION TITLE
Lesson 4: Import "solcx" could not be resolved

### DISCUSSION THREAD

**Discussion by Tokenomacs** [smartcontractkit/full-blockchain-solidity-course-py#816] [Q&A] [answered]
Hi guys!

Unfortunately, I have been struggling with the error in the header for a while now.

I have tried two solutions found online which don't seem to work for me:

1. Importing a virtual environment in same directory as web3_py_simple_storage.py

2. Following the two fixes as listed here: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md (and have even directly copied source code from here: https://github.com/PatrickAlphaC/web3_py_simple_storage/blob/main/deploy.py)

Attached is a picture of my code. I have also used pip install to install both pip3 install py-solc-x and pip install web3.

Anyone have any solutions they found? Would be much appreciated so I can follow along with video :)

![image](https://user-images.githubusercontent.com/92122220/149997319-d9e5c7fa-02fd-4e82-9540-9680d0aa5320.png)

**Comment by cromewar** [comment]
Hello @Tokenomacs  it seems you are using python with conda, and it is installed on the `C` which requires elevated privileges, I recommend one of this alternatives:

1. use normal python, install it from it's page not using conda.
2. use WSL (windwos subsystem for linux) and manage all your web3/python development there.

**Reply by Tokenomacs** [reply]
Hi there! 

Thanks so much for your help! Interestingly after I saw this error I uninstalled anaconda, however still received the same error specifically with the solcx import line.

Could this be possible? Or do you believe there is still interference from my previous management with Anaconda python?

**Reply by cromewar** [reply]
Yes it should, also remember you have to choose the python interpreter on vscode (bottom left corner), and select the new one you have installed.

**Reply by shucks18** [reply]
oh my days it was the python interpreter for me thanks for this!!

**Comment by Tokenomacs** [comment]
Hi Vitaly thanks for all your help!

 

I recently solved the solcx issue, but am still having trouble with the web3 import.



It is installed as seen by the requirement already satisfied in the terminal.

 

Any diagnosis?

 

Thanks so much

**Comment by Tokenomacs** [comment]
Hey Shucks!

First I would copy the updated code as provided in the GitHub repo associated with lesson 4.

Next I would make sure that I have solcx installed through pip.

Then, the trick for me was realizing that I needed to select the correct python interpreter as specified above (thanks Vitaly!)

Let me know if this solves your problem

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Tokenomacs** [comment]
Nice! It seems like a common problem

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by cromewar** [comment, marked as answer]
Closing this thread as this is solved now:
Solution: select the correct python interpreter on vscode.
